contributing/chestertons-fence.md

```diff
@@ -0,0 +1,79 @@
+# Chesterton's Fence
+
+Hello :wave:
+You're probably here becomes someone directed you to this post in response to you suggesting an idea about how Exercism could be improved.
+Before our team invest time into exploring your idea, they're hoping you'll read this and consider whether your idea might of being thought of before, what pitfalls it might contain.
+**By adding those problems and pitfalls to your suggestion, and considering why your idea might not have already been implemented, you are likely to get a faster and more positive response.**
+
+## So what does "Chesterton's Fence" mean?
+
+Chesterton’s Fence is an idea inspired by a quote from the writer G.K. Chesterton’s 1929 book, The Thing.
+It became well-known as it was quoted by John F. Kennedy.
+This is the original quote:
+
+> There exists in such a case a certain institution or law; let us say, for the sake of simplicity, a fence or gate erected across a road.
+> The more modern type of reformer goes gaily up to it and says, “I don’t see the use of this; let us clear it away.”
+> To which the more intelligent type of reformer will do well to answer: “If you don’t see the use of it, I certainly won’t let you clear it away. Go away and think. Then, when you can come back and tell me that you do see the use of it, I may allow you to destroy it."
+
+The idea of Chesterton’s Fence is that you if don't understand why something is there, you probably don't understand why it should be removed.
+Equally, if you don't understand why something **isn't** there, you probably don't understand why it's been left out.
+
+A simple rule to remember is: **"Do not remove a fence until you know why it was put up in the first place."**
+
+## This Notion is Relevant to Exercism how?
+
+Exercism has been designed, engineered and built by thousands of very talented individuals.
+Nearly everything with Exercism has been debated, discussed and rewritten many times.
+Exercism is a very intentional product - things are there because they've been designed to be there, and things are often left out because they've been designed to be left out.
+
+That said, every day someone comes up with a great idea that we've not thought of before - an idea that's new or better, or unblocks our thinking.
+So if you have an idea or a suggestion, we welcome it!
+
+But...
+
+Before posting it, please ask yourself whether we might have already considered it, and why we might have done things differently to what you're suggesting.
+And remember, the more obvious your idea seems - the more likely it's been discussed and debated many times over -- so when you present it, present it with the potential caveats and pitfalls in place.
+If you have an instinct to say "why not **just** ...", then it almost certainly falls into this category.
```

Comment:
Perhaps this could also suggest doing a quick search through issues. I think if there are no results, it's fair to ask about the reasons. That issue would then also serve as future documentation of the decision.

Reply:
Yeah, that's good point. However, I think I'll add something about reading other issues to a different doc - I want to keep this doc about the thought process one should use, rather than the workflow.